File: features/top_ads/top_ads_dashboard/src/main/java/com/tokopedia/topads/dashboard/domain/repository/TopAdsDashboardRepository.java
package com.tokopedia.topads.dashboard.domain.repository;

import com.tokopedia.topads.dashboard.data.model.DataCredit;
import com.tokopedia.topads.dashboard.data.model.DataStatistic;
import com.tokopedia.topads.dashboard.data.model.TotalAd;
import com.tokopedia.usecase.RequestParams;

import java.util.List;

import rx.Observable;

/**
 * Created by nakama on 24/04/18.
 */

public interface TopAdsDashboardRepository {

    Observable<TotalAd> populateTotalAds(RequestParams requestParams);

    Observable<DataStatistic> getStatistics(RequestParams requestParams);

    Observable<List<DataCredit>> getDashboardCredit(RequestParams requestParams);
}

File: features/top_ads/top_ads_dashboard/src/main/java/com/tokopedia/topads/dashboard/domain/repository/TopAdsDashboardRepository.kt
package com.tokopedia.topads.dashboard.domain.repository

import com.tokopedia.topads.dashboard.data.model.CreditResponse
import com.tokopedia.topads.dashboard.data.model.DataCredit
import com.tokopedia.topads.dashboard.data.model.DataStatistic
import com.tokopedia.topads.dashboard.data.model.TotalAd
import com.tokopedia.usecase.RequestParams

import rx.Observable

/**
 * Created by nakama on 24/04/18.
 */

interface TopAdsDashboardRepository {

    fun getStatistics(requestParams: RequestParams): Observable<DataStatistic>

}
